ratis-test/src/test/java/org/apache/ratis/datastream/TestDataStream.java

```diff
@@ -97,33 +98,44 @@ public WritableByteChannel getWritableByteChannel() {
 
     @Override
     public CompletableFuture<DataStream> stream(RaftClientRequest request) {
-      writeRequest = request;
+      if (request.getClientId().equals(impl.getHeader().getClientId())) {
```

Comment:
why we need this check ?

Reply:
Because all statemachine will share `writeRequest` thus set it.

Reply:
you can put writeRequest in SingleDataStreamStateMachine as a field.